pkg/operator/options/options.go

```diff
@@ -57,23 +83,113 @@ func New() *Options {
 	f.BoolVar(&opts.EnableLeaderElection, "leader-elect", env.WithDefaultBool("LEADER_ELECT", true), "Start leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
 	f.Int64Var(&opts.MemoryLimit, "memory-limit", env.WithDefaultInt64("MEMORY_LIMIT", -1), "Memory limit on the container running the controller. The GC soft memory limit is set to 90% of this value.")
 
+	// Vars that must be merged with settings
+	f.DurationVar(&opts.BatchMaxDuration, "batch-max-duration", env.WithDefaultDuration("BATCH_MAX_DURATION", 10*time.Second), "TODO")
+	f.DurationVar(&opts.BatchIdleDuration, "batch-idle-duration", env.WithDefaultDuration("BATCH_IDLE_DURATION", time.Second), "TODO")
+	f.StringVar(&opts.featureGates, "feature-gates", env.WithDefaultString("FEATURE_GATE", ""), "TODO")
+
 	if opts.MemoryLimit > 0 {
 		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
 		debug.SetMemoryLimit(newLimit)
 	}
+
 	return opts
 }
 
-// MustParse reads the user passed flags, environment variables, and default values.
-// Options are valided and panics if an error is returned
-func (o *Options) MustParse() *Options {
-	err := o.Parse(os.Args[1:])
-
-	if errors.Is(err, flag.ErrHelp) {
+func (*Options) Inject(ctx context.Context) (context.Context, error) {
```

Comment:
Any reason to change this from `MustParse`

Reply:
Following the `Injectable` pattern, will change back since we're dropping that.